src/go/rpk/pkg/config/weak_test.go

```diff
@@ -0,0 +1,429 @@
+package config
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"gopkg.in/yaml.v3"
+)
+
+func TestWeakBool(t *testing.T) {
+	type testWeakBool struct {
+		Wb weakBool `yaml:"wb"`
+	}
+
+	tests := []struct {
+		name    string
+		data    string
+		expBool bool
+		expErr  bool
+	}{
+		{
+			name:    "should unmarshal boolean:true",
+			data:    "wb: true",
+			expBool: true,
+			expErr:  false,
+		},
+		{
+			name:    "should unmarshal boolean:false",
+			data:    "wb: false",
+			expBool: false,
+			expErr:  false,
+		},
+		{
+			name:    "should unmarshal int:0",
+			data:    "wb: 0",
+			expBool: false,
+			expErr:  false,
+		},
+		{
+			name:    "should unmarshal int:different from zero",
```

Comment:
`different from zero` is a bit confusing on the first read, how about `should unmarshal non-zero int`?

Also fwiw I think the "should unmarshal" aspect in these tests makes the testing output more verbose and doesn't add value, for example here it could be `non-zero int`